### ISSUE TITLE
`bentoml get` fails for one-off "save to s3" bundle

### ISSUE THREAD

**Issue by yubozhao** [bentoml/BentoML#698]
**Describe the bug**

```bash
$ bentoml get BENTO_TAG

BentoService BENTO_TAG not found - INTERNAL:Bento BENTO_TAG not found in target repository
```

**To Reproduce**
1. Caa BentoService#save with s3 path
```
# update last line in guide/quick-start/main.py to 
# saved_path = iris_classifier_service.save('s3://my-test-bucket/')
$ python guide/quick-start/main.py
```
2. run `bentoml get IrisClassifier:latest`

**Expected behavior**
Expect CLI to display relevant information for the saved BentoService

**Environment:**
 - OS: [e.g. MacOS 10.15.5]
 - Python/BentoML Version [e.g. Python 3.7.3, BentoML-0.7.7]

**Comment by yubozhao** [comment]
Problem is CLI will initialize a YataiService with LocalBentoRepo, when getting information for a BentoService using s3 storage. It will use the LocalBentoRepo instead of the S3 one

**Comment by parano** [comment]
Note: `.save("s3://..")` is a legacy behavior that we plan to deprecate, we should actually avoid saving it to the DB as well. The YataiService is designed to use with only one type of storage system. It would be nice to give the user a better error message.

We should probably add a warning to `BentoService#save` method, when the `path` is provided there, tell the user:
```python
logger.warning("""
Executing one-off save to path '{path}', the saved bundle can be loaded with `bentoml.load(path)`
and API server can be started with CLI command `bentoml serve {path}`. But the saved bundle
will not appear in the `bentoml list` command output and will not be stored in the YataiService model registry.
""")
```

**Comment by parano** [comment]
@yubozhao would be great to think about this in the upcoming YataiService push/pull related changes

**Comment by parano** [comment]
cc @yubozhao 

1. Recommend `save(yatai_url=...)`, raise exception when using `save(path)`, ask user to use `save_to_dir` instead
2. `save_to_dir(file_path)` 👈 add support for s3 path

**Comment by parano** [comment]
> cc @yubozhao
> 
> 1. Recommend `save(yatai_url=...)`, raise exception when using `save(path)`, ask user to use `save_to_dir` instead
> 2. `save_to_dir(file_path)` 👈 add support for s3 path

this is now implemented in the current version of BentoML 0.12.1
